### DISCUSSION TITLE
org-level dependencyDashboard

### DISCUSSION THREAD

**Discussion by travi** [renovatebot/renovate#9789] [Suggest an Idea]
i'm curious if the idea of an org-level dependency dashboard in some form has been discussed before. i think a way to provide visibility beyond individual repos would help sell the value of renovate even further within an enterprise setting. 

it could be very helpful to provide visibility around pending PRs for updates related to node version or security related dependency updates to teams like a technical leadership team or a security team. that could enable them to follow up with the teams responsible for the individual repos and encourage them to follow through with merging important updates that would otherwise require more detailed digging through each repo individually.

assuming something along these lines isnt already available in a way that i've overlooked, has this type of tool been considered? any chance there are plans along these lines that i could share with my team?

**Comment by rarkins** [comment]
It's definitely something we'd like to do, but no concrete plans yet. We've been tending to focus our energies on the core open source tool and the dashboard part hasn't been improved in quite a while.

If you had ideas for how you would anticipate the UX to work, that would certainly help move things forward. e.g. considering the scenarios where people have hundreds of repos, or thousands of PRs, or divergent versions of the same dependency, then it gets hard to think of a "simple" way to present things.

**Reply by travi** [reply]
certainly understandable. i'll try to give a high-level idea of what got me down the path of thinking about this. happy to clarify any thoughts that might be helpful, especially as you get closer to thinking through implementation a little more closely. for the most part, i would see a solution like this as a tool that is useful for teams that are a little more removed from the implementation teams that are maintaining individual projects.

in an enterprise setting, a great example is the security team that is interested in making sure a vulnerability is patched in _all_ projects that use the dependency. in that scenario, a view that enabled understanding the list of projects that had outstanding PRs for upgrading to a patched version. this seems like one of those scenarios where divergent versions come into play. a security team is likely less interested in how out of date a project is or even whether they are using the latest version. they want the answer of whether the project has updated beyond the fixed version. i would expect that they would find outstanding PRs for newer versions than the fixed version as noisy and would prefer those be filtered away.

another scenario that comes to mind could be when an infrastructure team would like all production applications to be on the same version of the language. for node apps, there may be a desire to delay updating to a new major when the lts tag switches from 14 to 16, so the teams may be asked to delay merging. once the infra team is comfortable with teams moving forward, they would like all teams to make the switch fairly quickly. having a place to see that progress would help them coordinate. having a quick way to get from that view to an actionable way to approve or merge each update (a simple hyperlink to the PR is probably enough) could also be useful, depending on how that type of responsibility is distributed among the teams.

i'm not sure i would see a lot of value in an overall dashboard of all outstanding upgrades, but i think being dynamic enough to filter to this level or defining specific workflows like these would be extremely useful. i'm sure i could come up with some other scenarios, but mainly i'm thinking about the cases where similar changes are desired in a consistent way across many repos and there is a desire for certain versions to either all be the same or all be passed a certain threshold. this doesn't apply to the majority of dependencies, but special scenarios like these do come up here and there.

**Reply by travi** [reply]
the other scenario that i should have included here would be around internal libraries. consider a component library distributed to various applications as a package. the design team may be interested in the rollout of certain visual changes being fairly synchronized. most updates may not matter much for timing, but there could be certain releases where they want to help coordinate the merging and deployment, so visibility to whether the PRs for those specific versions are merged would help them with that.

**Reply by rarkins** [reply]
On a related note, I've thought about whether "allowedVersions" on a org-wide scale would be useful, e.g. to control when a particular version can be updated.

**Reply by travi** [reply]
yes, i think that could be a very valuable feature. 

i have been in conversations along these lines for language version especially. even once a version gets to LTS status, there may be a desire to "bless" that version, so it would be preferred if the upgrade did not happen ahead of the team having a chance to do their review. once the version is approved, there is a desire for all projects to migrate fairly quickly and then not allow the older version(s) anymore. in a case like this, i assume this would be defined in a shareable preset, which i think could benefit from a PR that adds the new version to the list of allowed versions.

while the other scenarios would be less likely to have a desire to delay upgrades, they would likely benefit from being able to define a minimum allowed value at least in order to define a filter for the tools that could encourage folks to complete pending upgrades, like an org dashboard.

**Reply by rknj** [reply]
Hello there. Any news on this topic? I also see it very useful for us.
